src/pagination/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import theme from '@dojo/framework/core/middleware/theme';
+import { WNode, RenderResult } from '@dojo/framework/core/interfaces';
+import { createResource, DataTemplate } from '@dojo/framework/core/resource';
+
+import Icon from '../icon';
+import Select, { defaultTransform } from '../select';
+
+import bundle from './Pagination.nls';
+import * as css from '../theme/default/pagination.m.css';
+
+function createMemoryTemplate<S = void>(): DataTemplate<S> {
+	return {
+		read: (_, put, get) => {
+			let data: any[] = get();
+			put(0, data);
+			return { data, total: data.length };
+		}
+	};
+}
+
+const memoryTemplate = createMemoryTemplate();
+
+export interface PaginationProperties {
+	/** The initial page number */
+	initialPage?: number;
+
+	/** The initial page size */
+	initialPageSize?: number;
+
+	/** Callback fired when the page number is changed */
+	onPageChange(page: number): void;
```

Comment:
How about renaming these callbacks to `onPage` and `onPageSize` as we've done out best to drop `onChange` etc from elsewhere.